### PR TITLE
Add sourcemap periodic refresh tracing

### PR DESCRIPTION
**Pull request by dmathieu**
## Motivation/summary

Sourcemap periodic refreshes are making external calls to ES. Those calls produce spans, but they can't generate any data, as there is no parent transaction.
This creates a transaction for each sync run, so we can observe them.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [x] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

Outside of unit tests, I have been testing these changes by running apm-server with tilt and rum/self instrumentation enabled.

Screenshot 2023-06-30 at 09 42 52

I'm not 100% sure about the `initialSearch` and `scrollSearch` spans, since they do nothing but wrap the ES call. They also help in figuring out which method makes those calls though.

## Related issues

See #1014.

### CODE REVIEW

**Comment by dmathieu** [comment]
System tests failed once du to a flaky test. They passed on a second run, but somehow, the `System Tests Results` is still marked as failed. And rerunning it doesn't change anything.

File: changelogs/head.asciidoc [hunk 1]
```diff
@@ -17,3 +17,4 @@ https://github.com/elastic/apm-server/compare/8.9\...main[View commits]
 
 [float]
 ==== Added
+- Add a transaction to the source map periodic refresh action. {pull}11116[11116]
```
Comment (carsonip) [inline changelogs/head.asciidoc:20]: Would be great to mention this is for instrumentation only.

**Review by carsonip — APPROVED** [review]

File: changelogs/head.asciidoc (hunk 1 above)
Comment (dmathieu) [reply changelogs/head.asciidoc]: How about now?

**Review by carsonip — APPROVED** [review]

**Comment by axw** [comment]
Verified with 8.10.0-BC1. Running apm-server under Tilt, I configured `instrumentation.enabled: true` and `apm-server.rum.enabled: true` and observed the "MetadataESFetcher.sync" traces being captured:

image

I think the transaction name could mention sourcemaps somewhere, but this is good enough for now.

**Pull request merged**